content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,153 @@
+
+## Network security
+
+A number of [Container Network Interface (CNI) plugins](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
+plugins provide the functionality to
+restrict network resources pods may communicate with. This is most commonly done
+through [Network Policies](/docs/concepts/services-networking/network-policies/)
+which provide a namespaced resource to define rules.
+
+Not all CNI plugins provide encryption in transit. If the chosen plugin lacks this
+feature, an alternative solution could be to use a service mesh to provide that
+functionality.
+
+The `etcd` datastore of the control plane should have controls to limit access.
+Furthermore, mutual TLS (mTLS) should be used to communicate securely with it.
+The certificate authority for this should be unique to `etcd`.
+In addition, external Internet access to the Kubernetes API server should be restricted
+to not expose the API publicly. You can then use a bastion host to access the server.
+If a cloud provider is used for hosting Kubernetes, the access from pods to the cloud
+metadata API `169.254.169.254` should also be restricted or blocked if not needed
+because it may leak information.
+
+## Pod security
+
+RBAC authorization is crucial but
+[cannot be specific enough on the Pods resource](/docs/concepts/security/rbac-good-practices/#workload-creation)
+(or on any resource that manages Pods). The only granularity is the API verbs
+on the resource itself, for example `create` on Pods. Indeed, without
+additional admission, the authorization to create these resources allows direct
+unrestricted access to on the schedulable nodes of a cluster.
+
+[Pod Security Admission](/docs/concepts/security/pod-security-admission/) is an
+admission controller to enable fine-grained authorization of Pod creation and
+updates. It replaces the Pod Security Policy admission controller. 
+The [Pod Security Standards](/docs/concepts/security/pod-security-standards/)
+define three different policies that restrict how fields can be set in the `PodSpec`.
+For a hands-on tutorial on Pod Security, see the blog post
+"[Kubernetes 1.23: Pod Security Graduates to Beta](/blog/2021/12/09/pod-security-admission-beta/)".
+
+{{< note >}}
+[PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)
+will be removed in version 1.25.
+{{< /note >}}
+Pod Security admission `restricted` policy can enforce or even warn of the part of the
+[security context](/docs/tasks/configure-pod-container/security-context/) that must
+be properly set. Nevertheless, pods' [security context](/docs/tasks/configure-pod-container/security-context/)
+should be separately investigated to limit the privileges and access pods may have.
+
+[Memory and CPU limits](/docs/concepts/configuration/manage-resources-containers/)
+should be set in order to restrict the memory and CPU resources a pod can consume on
+a node, and therefore prevent potential DoS attacks.
+
+## Enabling seccomp
```

Comment:
If this is a checklist, we should bear in mind that:
- not all clusters would let you enable seccomp
- “enabling” isn't something you check off; you would check off “seccomp is enabled and in use for all Linux nodes” or similar.

Think about whether we want to insist on _both_ seccomp and AppArmor, or if we want to insist on at least one of those mechanisms.

Reply:
Maybe we could add an *Additional Security* section for this? :thinking:

Reply:
It seems to me that seccomp and AppArmor are different mechanisms with different purposes. Like you cannot block syscalls with AppArmor and that's the whole point of seccomp(-BPF). So ideally you can use both of them.

Reply:
Windows nodes :wink:
We shouldn't make a checklist that accidentally always fails any cluster that only uses Windows nodes.

Reply:
> Also, I'd make this a 3rd level heading: seccomp is part of Pod security.

We discussed that a little bit here https://github.com/kubernetes/website/pull/33992#discussion_r885587185. I agree that from a technical point of view, it could belong to the pod security, as well as AppArmor (even if it only relies on annotations for now).

Reply:
I added a note to precise that these two are Linux-only and that AppArmor is even more restricted. I also moved them into Pod Security, since there is a KEP going on to finally make AppArmor GA and put its annotation in a PodSecurityContext field, so it should make sense! It's here https://github.com/kubernetes/website/pull/33992/commits/9999e18274a6306733f073437bb9eae17d8e3430.